display/man.rs

```
@@ -0,0 +1,445 @@
+            Ok(formatted_man_page) => return Ok(formatted_man_page),
+            Err(err) if err.kind() == io::ErrorKind::NotFound => continue,
+            Err(err) => return Err(err.into()),
+        }
+    }
+
+    Err(ManError(
+        "neither groff(1), nor nroff(1), nor mandoc(1) are installed".to_string(),
+    ))
+}
+
+/// Formats man page content into appropriate format.
+///
+/// # Arguments
+///
+/// `man_page` - [Vec<u8>] with content that needs to displayed.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [std::io::Error] if failed to execute pager or failed write to its STDIN.
+fn display_pager(man_page: Vec<u8>) -> Result<(), io::Error> {
+    let pager = std::env::var("PAGER").unwrap_or_else(|_| "more".to_string());
+
+    let args = if pager.ends_with("more") {
+        vec!["-s"]
+    } else {
+        vec![]
+    };
+
+    spawn(&pager, args, Some(&man_page), Stdio::inherit())?;
+
+    Ok(())
+}
+
+/// Displays man page
+///
+/// # Arguments
+///
+/// `name` - [str] name of system documentation.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [ManError] if man page not found, or any display error happened.
+fn display_man_page(name: &str) -> Result<(), ManError> {
+    let cat_output = get_man_page(name)?;
+    let formatter_output = format_man_page(cat_output)?;
+    display_pager(formatter_output)?;
+
+    Ok(())
+}
+
+/// Displays man page summaries for the given keyword.
+///
+/// # Arguments
+///
+/// `keyword` - [str] name of keyword.
+///
+/// # Returns
+///
+/// Nothing
+///
+/// # Errors
+///
+/// [ManError] if call of `apropros` utility failed.
+fn display_summary_database(keyword: &str) -> Result<(), ManError> {
+    let output: Output = Command::new("apropos").arg(keyword).output()?;
+
+    if !output.status.success() {
+        return Err(ManError("apropos command failed".to_string()));
+    }
+
+    let result = String::from_utf8_lossy(&output.stdout);
```

Redundant buffer. If this was done to hide a possible error from `apropos` - this is unhelpful. Just use `spawn` instead of `output` [so the stdout is inherited](https://doc.rust-lang.org/std/process/struct.Command.html#method.stdout)